helm: ensure that envoy daemonset is installed only when needed

f1ko opened the pull request:
As `.Values.ingressController.enabled` and `.Values.gatewayAPI.enabled` require `.Values.l7Proxy` to be set it's sufficient to just perform a check on that value.

<!-- Description of change -->

Fixes https://github.com/cilium/cilium/issues/33430

No backport necessary as this only affects 1.16

```release-note
helm: ensure that envoy daemonset is installed only when needed
```

sayboras commented on install/kubernetes/cilium/templates/_helpers.tpl:
```diff
@@ -145,7 +145,9 @@ Enable automatic lookup of k8sServicePort from the cluster-info ConfigMap (kubea
 */}}
 {{- define "envoyDaemonSetEnabled" }}
-  {{- if (not (kindIs "invalid" .Values.envoy.enabled)) }}
+  {{- if (eq .Values.l7Proxy false) }}
```
nit: probably just `not  .Values.l7Proxy` instead

sayboras reviewed (approved):
ah nice, thanks a lot. I forgot  about this flag.

f1ko committed:
helm: ensure that envoy daemonset is installed only when needed

As `.Values.ingressController.enabled` and
`.Values.gatewayAPI.enabled` require
`.Values.l7Proxy` to be set it's sufficient to
just perform a check on that value.

Signed-off-by: Filip Nikolic <oss.filipn@gmail.com>
